### PR TITLE
Http logging basic auth

### PR DESCRIPTION
**Pull request by corylanou**
Too much work was being done at the logger level to determine basic auth username.  The standard library can already do what we need here.

### CODE REVIEW

**Comment by otoolep** [comment]
That's cool, @corylanou -- original code was copied from 0.8, but the if the libraries support it, *great*.

+1

**Comment by otoolep** [comment]
@corylanou -- seems we could change `parseCredentials()` too, no?

**Comment by corylanou** [comment]
@otoolep yeah, Iooked at `parseCredentials` once, but on another look it was clearly working too hard.  `BasicAuth()` works fine here as well.

File: httpd/handler.go [hunk 1]
```diff
@@ -350,27 +349,15 @@ func httpError(w http.ResponseWriter, error string, code int) {
...
+	if u, p := q.Get("u"), q.Get("p"); u != "" && p != "" {
+		return u, p, nil
 	}
-	fields = strings.Split(string(bs), ":")
-	if len(fields) != 2 {
-		return "", "", fmt.Errorf("invalid Basic Authentication value")
+	if u, p, ok := r.BasicAuth(); ok {
+		return u, p, nil
+	} else {
+		return "", "", fmt.Errorf("unable to parse username/password from header")
```
Comment (otoolep) [inline httpd/handler.go]: Minor: could you make this message more precise e.g. "unable  to parse Basic Auth credentials"?

**Comment by otoolep** [comment]
Great change, 1 comment on the error message. +1.

**Pull request merged**